src/opcua-iiot-result-filter.ts

```diff
@@ -264,26 +284,37 @@ module.exports = (RED: nodered.NodeAPI) => {
       return result
     }
 
-    const filterByReadType = (payload: Todo) => {
-      return {
-        value: payload.value.filter((item: Todo) => {
-          return item.nodeId.toString().includes(this.nodeId)
-        })
+    const filterByReadType = (payload: ReadPayload) => {
+      if (isArray(payload.value))
+        return {
+          value: payload.value.filter((item: Todo) => {
```

Comment:
the `: Todo` after `item` should probably be removed. See [this comment](https://github.com/DATATRONiQ/node-red-contrib-iiot-opcua/pull/10/files#r878138426)

Reply:
TypeScript fails to understand that the isArray check applies to the same object as the .filter() call.